algebra/tests/matrix/basic_interactions.c

```diff
@@ -161,3 +174,156 @@ TEST_GROUP_RUNNER(group_matrix_at)
+
+
+TEST_SETUP(group_matrix_bufAlloc)
+{
+}
+
+
+TEST_TEAR_DOWN(group_matrix_bufAlloc)
+{
+	free(dynMat.data);
+	dynMat.data = NULL;
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_structElems)
+{
+	dynMat.rows = 0;
+	dynMat.cols = 0;
+	dynMat.transposed = 1;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
+
+	TEST_ASSERT_EQUAL_INT(ROWS, dynMat.rows);
+	TEST_ASSERT_EQUAL_INT(COLS, dynMat.cols);
+
+	TEST_ASSERT_FALSE(dynMat.transposed);
+
+	TEST_ASSERT_NOT_NULL(dynMat.data);
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_validSeek)
+{
+	int row, col;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
+
+	for (row = 0; row < ROWS; row++) {
+		for (col = 0; col < COLS; col++) {
+			TEST_ASSERT_NOT_NULL(matrix_at(&dynMat, row, col));
+		}
+	}
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_invalidSeek)
+{
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
+
+	algebraTests_checkInvalidSeek(&dynMat);
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_initVal)
+{
+	int row, col;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
+
+	for (row = 0; row < ROWS; row++) {
+		for (col = 0; col < COLS; col++) {
+			TEST_ASSERT_EQUAL_FLOAT(0, *matrix_at(&dynMat, row, col));
+		}
+	}
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_readAndWrite)
+{
+	int row, col, i;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_SUCC, matrix_bufAlloc(&dynMat, ROWS, COLS));
+
+	/* Assigning different value to every matrix element */
+	i = 0;
+	for (row = 0; row < ROWS; row++) {
+		for (col = 0; col < COLS; col++) {
+			*matrix_at(&dynMat, row, col) = i;
+			i++;
+		}
+	}
+
+	/* Checking every matrix element */
+	i = 0;
+	for (row = 0; row < ROWS; row++) {
+		for (col = 0; col < COLS; col++) {
+			TEST_ASSERT_EQUAL_FLOAT(i, *matrix_at(&dynMat, row, col));
+			i++;
+		}
+	}
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_zeroRows)
+{
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, COLS));
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, -COLS));
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_zeroCols)
+{
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, ROWS, 0));
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, -ROWS, 0));
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_zeroRowsAndCols)
+{
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, 0));
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_overflow1)
+{
+	int rows = sqrt(INT_MAX);
+	int cols = rows;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, rows, cols));
+}
+
+
+TEST(group_matrix_bufAlloc, matrix_bufAlloc_overflow2)
+{
+	/* This checks if matrix_buAlloc is ready for potential overflow while calculating rows * cols  */
+	int rows = INT_MAX;
+	int cols = INT_MAX;
+
+	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, rows, cols));
+}
```

Comment:
```suggestion
TEST(group_matrix_bufAlloc, matrix_bufAlloc_invalidArgs)
{
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, 0));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, 0, COLS));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, ROWS, 0));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, sqrt(UINT_MAX), sqrt(UINT_MAX)));
	TEST_ASSERT_EQUAL_INT(BUFF_ALLOC_FAIL, matrix_bufAlloc(&dynMat, UINT_MAX, UINT_MAX));
}
```

For small amount of arguments edge/corner cases this might be more readable